File: src/pytra/utils/browser/widgets/dialog.py
"""browser.widgets.dialog shim."""

from __future__ import annotations


class Dialog:
    def __init__(self, title: str = "", ok_cancel: bool = False) -> None:
        self.title = title
        self.ok_cancel = ok_cancel


class EntryDialog:
    def __init__(self, title: str = "", value: str = "") -> None:
        self.title = title
        self.value = value


class InfoDialog:
    def __init__(self, title: str = "", text: str = "") -> None:
        self.title = title
        self.text = text

File: src/runtime/cpp/pytra-gen/utils/browser/widgets/dialog.cpp
#include "runtime/cpp/pytra/built_in/py_runtime.h"

#include "pytra/utils/browser/widgets/dialog.h"


namespace pytra::utils::browser::widgets::dialog {

    /* browser.widgets.dialog shim. */
    
    
    struct Dialog : public PyObj {
        bool ok_cancel;
        str title;
        
        Dialog(const str& title = "", bool ok_cancel = false) {
            this->title = title;
            this->ok_cancel = ok_cancel;
        }
    };
    
    struct EntryDialog : public PyObj {
        str title;
        str value;
        
        EntryDialog(const str& title = "", const str& value = "") {
            this->title = title;
            this->value = value;
        }
    };
    
    struct InfoDialog : public PyObj {
        str text;
        str title;
        
        InfoDialog(const str& title = "", const str& text = "") {
            this->title = title;
            this->text = text;
        }
    };
    
}  // namespace pytra::utils::browser::widgets::dialog
